### PR TITLE
Validate the git path while registering new applicaiton and refactor stage log

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/api/api/web_api.go [hunk 1]
```diff
@@ -318,6 +319,12 @@ func (a *WebAPI) AddApplication(ctx context.Context, req *webservice.AddApplicat
 		a.logger.Error("failed to extract claims", zap.Error(err))
 		return nil, status.Error(codes.Internal, "internal error")
 	}
+
+	// The path to the application directory must be relative.
+	if strings.HasPrefix(req.GitPath.Path, "/") {
+		return nil, status.Error(codes.InvalidArgument, "the path must be a relative path")
+	}
```
Comment (nghialv) [inline pkg/app/api/api/web_api.go:326]: Some entities in our database are prefixed by this `/`.

File: pkg/app/piped/logpersister/stagelogpersister.go [hunk 2]
```diff
@@ -50,12 +50,9 @@ type stageLogPersister struct {
...
 
-	// We also send this log messages to the local logger.
-	switch s {
-	case model.LogSeverity_ERROR:
-		sp.logger.Error(log)
-	default:
-		sp.logger.Info(log)
+	// We also send the error logs to the local logger.
+	if s == model.LogSeverity_ERROR {
+		sp.logger.Warn(fmt.Sprintf("STAGE ERROR LOG: %s", log))
```
Comment (nghialv) [inline pkg/app/piped/logpersister/stagelogpersister.go:55]: Send all stage logs to the piped's logger is annoying. So I changed to send only the ERROR log.

File: pkg/app/api/api/web_api.go (hunk 1 above)
Comment (nghialv) [reply pkg/app/api/api/web_api.go:326]: That is why some applications were not triggered even if their files were touched.

**Review by nakabonne — APPROVED** [review]
Pretty neat. We have to consider more unexpected cases.

**Pull request merged**